site/_posts/2023-04-13-securing-react-native-with-oauth-0-71.md

```diff
@@ -0,0 +1,501 @@
+    <array>
+      <string>fusionauth.demo</string>
+    </array>
+  </dict>
+</array>
+```
+
+Here, the `CFBundleURLSchemes` defines the URL schemes handled by this application. The scheme we are registering, `fusionauth.demo`, should look familiar, as we configured FusionAuth to redirect to a URL with that scheme in it. If you modify it here, you should modify it there as well.
+
+The last step is to change `ios/RNfusionauth/AppDelegate.h` and `ios/RNfusionauth/AppDelegate.mm` to include needed imports and properties:
+
+AppDelegate.h:
+```objective_c
+{% remote_include 'https://raw.githubusercontent.com/sonderformat-llc/fusionauth-example-react-native-0-71/main/ios/RNfusionauth/AppDelegate.h' %}
+```
+
+AppDelegate.mm:
+```objective_c
+{% remote_include 'https://raw.githubusercontent.com/sonderformat-llc/fusionauth-example-react-native-0-71/main/ios/RNfusionauth/AppDelegate.mm' %}
+```
+
+The changes in `AppDelegate.mm` allows the authentication with [Magic Links](/docs/v1/tech/passwordless/magic-links).
+
+### Setting up auth for Android
+
+For Android, we need additional configuration to capture the [authorization redirect](https://github.com/openid/AppAuth-android#capturing-the-authorization-redirect). Add the following property to the `defaultConfig` object in the `android/app/build.gradle` file:
+
+```gradle
+android {
+ defaultConfig {
+   manifestPlaceholders = [
+     appAuthRedirectScheme: 'fusionauth.demo'
+   ]
+ }
+}
+```
+
+Here, the `appAuthRedirectScheme`, `fusionauth.demo`, is another scheme definition, the same as with iOS. If you modify it here, you should modify it in the FusionAuth administrative interface and in the iOS configuration as well.
+
+However, a new issue pops up when we start working on the Android application. Developing and debugging an Android app on a Mac is difficult as the emulator is not fully supported. Among other issues, the emulator is slow when compared to the iOS emulator. 
+
+A better solution is to use an actual Android mobile device. When you are doing so, how can you connect the FusionAuth server, running on localhost, to the device, which is on a Wi-Fi or cell network? The solution is to use a local tunnel service such as [ngrok](/docs/v1/tech/developer-guide/exposing-instance).
+
+## Coding a React Native application to use OAuth and FusionAuth
+
+Finally, the code!
+
+Big picture, we're going to be building out our logic and views in the `App.tsx` file. For a bigger project, you'd split this code up into components, but for our tutorial having one file will make things easier. We'll use libraries to manage authorization and secure storage of our data, however. 
+
+Here's what `App.tsx` will look like when we are done (don't worry, it looks like a lot, but we'll explain most of it):
+
+```tsx
+{% remote_include 'https://raw.githubusercontent.com/sonderformat-llc/fusionauth-example-react-native-0-71/main/App.tsx' %}
+```
+
+First, we need to add necessary imports to `App.tsx`:
+
```

Comment:
```suggestion

{%comment%}
TODO when we move to new blogging platform, have these fragments pulled out instead of hardcoded in here, because they'll decay if the content of `App.tsx` changes.
{%endcomment%}
```